baseapp/baseapp.go

```diff
@@ -653,9 +661,22 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte) (gInfo sdk.GasInfo, re
 		anteEvents = events.ToABCIEvents()
 	}
 
+	if mode == runTxModeCheck {
+		err = app.mempool.Insert(ctx, tx.(mempool.Tx))
+		if err != nil {
+			return gInfo, nil, anteEvents, priority, err
+		}
```

Comment:
What's the difference between this and lines 678-683 below ?

Reply:
Yeah, this doesn't look right @JeancarloBarrios. We should only be inserting into the app-side mempool on CheckTx once, which we already do below.

Reply:
Agree; sorry I have not fully updated this I will push a bit later today the implementation on likes 678-683has been removed.